### PR TITLE
Fix an invalid cast to MobileBrokerResponse

### PR DESCRIPTION
**Pull request by bgavrilMS**
Not an actual bug yet because we did not ship it. 

Issue was introduced by https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/commit/e1ec46204bd200399143d32e7bfe98e2ba4fdc0a

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Internal/AndroidBrokerTokenResponse.cs [hunk 1]
```diff
@@ -1,74 +1,74 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Internal/AndroidBrokerTokenResponse.cs:1]: Spacing... 
No change made to this file, only a rename to AndroidBrokerResponse.

File: src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs [hunk 2]
```diff
@@ -77,16 +77,8 @@ public async Task<MsalTokenResponse> FetchTokensAsync(CancellationToken cancella
 
             if (msalTokenResponse.Error != null)
             {
-                var brokerTokenResponse = (MobileBrokerTokenResponse)msalTokenResponse;
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs:80]: This cast for both iOS broker and Windows Broker.

File: src/client/Microsoft.Identity.Client/Internal/Requests/Silent/BrokerSilentStrategy.cs [hunk 3]
```diff
@@ -89,23 +89,8 @@ private async Task<MsalTokenResponse> SendTokenRequestToBrokerAsync()
...
-                    LogMessages.ErrorReturnedInBrokerResponse(msalTokenResponse.Error));
-
-                if (msalTokenResponse.Error == BrokerResponseConst.AndroidNoTokenFound ||
-                    msalTokenResponse.Error == BrokerResponseConst.AndroidNoAccountFound ||
-                    msalTokenResponse.Error == BrokerResponseConst.AndroidInvalidRefreshToken)
-                {
-                    throw new MsalUiRequiredException(msalTokenResponse.Error, msalTokenResponse.ErrorDescription);
-                }
-
-                var brokerTokenResponse = (MobileBrokerTokenResponse)msalTokenResponse;
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Internal/Requests/Silent/BrokerSilentStrategy.cs:102]: Same here, this cast is not valid

File: src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs [hunk 4]
```diff
@@ -66,26 +67,37 @@ private static bool IsThrottled(OAuth2ResponseBase oAuth2Response)
                oAuth2Response.ErrorDescription.StartsWith(Constants.AadThrottledErrorCode);
         }
 
-        internal static MsalServiceException FromBrokerResponse(
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs:69]: I refactored this method a bit. It does not need to receive so many params, just `msalResponse`

**Review by SameerK-MSFT — APPROVED** [review]
LGTM

**Review by SameerK-MSFT — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs (hunk 4 above)
Comment (SameerK-MSFT) [reply src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs:69]: Looks good

**Comment by pmaytak** [comment]
@SameerK-MSFT Since #3090 reverts TrueMAM code, these changes should be included into that separate TrueMAM feature branch instead of master for now.

**Pull request closed without merging**